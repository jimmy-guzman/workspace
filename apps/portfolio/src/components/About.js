import React from 'react'
import PropTypes from 'prop-types'
import about from '../data/about.json'

import TechList from './TechList'

const About = ({ isEnglish }) => (
  <main className='light'>
    <section className='about grid__row'>
      <div className='card grid__col--6'>
        <h1>{isEnglish ? 'About Me:' : 'Sobre Mi:'}</h1>
        <p>{isEnglish ? about.summary : about.summarySpanish}</p>
      </div>
      <div className='card grid__col--6'>
        <h1>{isEnglish ? 'Skills:' : 'Habilidades:'}</h1>
        <TechList techs={about.skills} />
      </div>
    </section>
  </main>
)

About.propTypes = {
  isEnglish: PropTypes.bool.isRequired
}

export default About
