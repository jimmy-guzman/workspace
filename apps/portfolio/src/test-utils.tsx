import React from 'react'
import { render, RenderOptions } from '@testing-library/react'
import { Router } from 'react-router-dom'
import { StyleSheetManager } from 'styled-components'
import { createMemoryHistory } from 'history'

import { ThemeProvider } from './providers'

const Providers: React.ComponentType<{ route?: string }> = ({
  children,
  route,
}: {
  route?: string
  children?: React.ReactNode
}) => {
  const history = createMemoryHistory()

  if (route) {
    history.push(route)
  }

  return (
    <StyleSheetManager disableVendorPrefixes>
      <Router history={history}>
        <ThemeProvider>{children}</ThemeProvider>
      </Router>
    </StyleSheetManager>
  )
}

Providers.displayName = 'ss'

const customRender = (
  ui: React.ReactElement,
  options?: Omit<RenderOptions, 'queries'>,
  route?: string
) => {
  const wrapper: React.ComponentType = ({
    children,
  }: {
    children?: React.ReactNode
  }) => <Providers route={route}>{children}</Providers>

  return render(ui, { wrapper, ...options })
}

export * from '@testing-library/react'

export { customRender as render }
