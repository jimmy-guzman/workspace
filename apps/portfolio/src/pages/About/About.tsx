import React from 'react'

import aboutLang from '@portfolio/lang/about.json'
import { TechList, Card } from '@portfolio/components'

export interface AboutProps {
  isEnglish?: boolean
}

export const About = ({ isEnglish = true }: AboutProps) => (
  <main className='light'>
    <section className='about grid__row'>
      <Card className='grid__col--6'>
        <h1>{isEnglish ? 'About Me:' : 'Sobre Mi:'}</h1>
        <p>{isEnglish ? aboutLang.summary : aboutLang.summarySpanish}</p>
      </Card>
      <Card className='grid__col--6'>
        <h1>{isEnglish ? 'Skills:' : 'Habilidades:'}</h1>
        <TechList technologies={aboutLang.skills} />
      </Card>
    </section>
  </main>
)
