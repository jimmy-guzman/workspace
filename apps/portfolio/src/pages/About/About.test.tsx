import React from 'react'

import { render } from '@portfolio/test-utils'
import { About, AboutProps } from './About'
import aboutLang from '@portfolio/lang/about.json'

const setupAbout = (props?: AboutProps) => render(<About {...props} />)

describe('<About />', () => {
  it('should render render english content', () => {
    const { getByText } = setupAbout()

    expect(getByText('About Me:')).toBeInTheDocument()
    expect(getByText(aboutLang.summary)).toBeInTheDocument()
    expect(getByText('Skills:')).toBeInTheDocument()
  })
  it('should render render spanish contennt', () => {
    const { getByText } = setupAbout({ isEnglish: false })

    expect(getByText('Sobre Mi:')).toBeInTheDocument()
    expect(getByText(aboutLang.summarySpanish)).toBeInTheDocument()
    expect(getByText('Habilidades:')).toBeInTheDocument()
  })
  it('should render render tech skills list', () => {
    const { getByText } = setupAbout()

    expect(getByText(aboutLang.skills[0])).toBeInTheDocument()
  })
})
