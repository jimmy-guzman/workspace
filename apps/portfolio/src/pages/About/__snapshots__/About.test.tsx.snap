// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<About /> should render default snapshot 1`] = `
<main
  class="light"
>
  <section
    class="about grid__row"
  >
    <div
      class="card grid__col--6"
    >
      <h1>
        Sobre Mi:
      </h1>
      <p>
        Hola, soy Jimmy Guzman y soy un Desarrollador de Web Front-End. En este momento estoy basado en Minneapolis. Estoy versado en varias habilidades de front-end y me apasiona la construcción de excelentes experiencias de usuario con gran funcionalidad, lo que se demuestra en mis proyectos
      </p>
    </div>
    <div
      class="card grid__col--6"
    >
      <h1>
        Habilidades:
      </h1>
      <ul
        class="TechList"
        style="text-align: center; list-style: none; padding: 0px;"
      >
        <li
          class="badge"
        >
          JavaScript
        </li>
        <li
          class="badge"
        >
          ReactJS
        </li>
        <li
          class="badge"
        >
          Redux
        </li>
        <li
          class="badge"
        >
          NodeJS
        </li>
        <li
          class="badge"
        >
          MongoDB
        </li>
        <li
          class="badge"
        >
          Webpack
        </li>
        <li
          class="badge"
        >
          Git
        </li>
        <li
          class="badge"
        >
          HTML
        </li>
        <li
          class="badge"
        >
          CSS
        </li>
        <li
          class="badge"
        >
          SASS
        </li>
        <li
          class="badge"
        >
          Bootstrap
        </li>
        <li
          class="badge"
        >
          jQuery
        </li>
        <li
          class="badge"
        >
          D3.js
        </li>
        <li
          class="badge"
        >
          Pug
        </li>
        <li
          class="badge"
        >
          Express
        </li>
        <li
          class="badge"
        >
          Firebase
        </li>
        <li
          class="badge"
        >
          Parcel
        </li>
        <li
          class="badge"
        >
          npm
        </li>
        <li
          class="badge"
        >
          Wordpress
        </li>
      </ul>
    </div>
  </section>
</main>
`;

exports[`<About /> should render snapshot with not isEnglish 1`] = `
<main
  class="light"
>
  <section
    class="about grid__row"
  >
    <div
      class="card grid__col--6"
    >
      <h1>
        Sobre Mi:
      </h1>
      <p>
        Hola, soy Jimmy Guzman y soy un Desarrollador de Web Front-End. En este momento estoy basado en Minneapolis. Estoy versado en varias habilidades de front-end y me apasiona la construcción de excelentes experiencias de usuario con gran funcionalidad, lo que se demuestra en mis proyectos
      </p>
    </div>
    <div
      class="card grid__col--6"
    >
      <h1>
        Habilidades:
      </h1>
      <ul
        class="TechList"
        style="text-align: center; list-style: none; padding: 0px;"
      >
        <li
          class="badge"
        >
          JavaScript
        </li>
        <li
          class="badge"
        >
          ReactJS
        </li>
        <li
          class="badge"
        >
          Redux
        </li>
        <li
          class="badge"
        >
          NodeJS
        </li>
        <li
          class="badge"
        >
          MongoDB
        </li>
        <li
          class="badge"
        >
          Webpack
        </li>
        <li
          class="badge"
        >
          Git
        </li>
        <li
          class="badge"
        >
          HTML
        </li>
        <li
          class="badge"
        >
          CSS
        </li>
        <li
          class="badge"
        >
          SASS
        </li>
        <li
          class="badge"
        >
          Bootstrap
        </li>
        <li
          class="badge"
        >
          jQuery
        </li>
        <li
          class="badge"
        >
          D3.js
        </li>
        <li
          class="badge"
        >
          Pug
        </li>
        <li
          class="badge"
        >
          Express
        </li>
        <li
          class="badge"
        >
          Firebase
        </li>
        <li
          class="badge"
        >
          Parcel
        </li>
        <li
          class="badge"
        >
          npm
        </li>
        <li
          class="badge"
        >
          Wordpress
        </li>
      </ul>
    </div>
  </section>
</main>
`;
